client/extensions/talk-to-plugin/state/reducer.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import { merge } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import {
+	OPTIONS_RECEIVE
+} from 'state/action-types';
+import { createReducer } from 'state/utils';
+
+const createItemsReducer = () => {
+	return ( state, { siteId, options } ) => {
+		return merge( {}, state, {
+			[ siteId ]: options
+		} );
+	};
+};
```

Comment:
Why not let the reducer itself live as a module constant, and then below directly use

```js
[ OPTIONS_RECEIVE ]: itemsReducer
```

? Rephrasing the question, why the reducer factory?